sockeye/data_io.py

```diff
@@ -656,50 +674,57 @@ def prepare_data(source_fnames: List[str],
                                                                                                max_seq_len_target)]
     logger.info("Buckets: %s", buckets)
 
-    # Pass 2: Randomly assign data to data shards
-    # no pre-processing yet, just write the sentences to different files
-    num_shards = get_num_shards(length_statistics.num_sents, samples_per_shard, min_num_shards)
-    logger.info("%d samples will be split into %d shard(s) (requested samples/shard=%d, min_num_shards=%d)."
-                % (length_statistics.num_sents, num_shards, samples_per_shard, min_num_shards))
-    shards, data_statistics = shard_data(source_fnames=source_fnames,
-                                         target_fnames=target_fnames,
-                                         source_vocabs=source_vocabs,
-                                         target_vocabs=target_vocabs,
-                                         num_shards=num_shards,
-                                         buckets=buckets,
-                                         length_ratio_mean=length_statistics.length_ratio_mean,
-                                         length_ratio_std=length_statistics.length_ratio_std,
-                                         output_prefix=output_prefix)
-    data_statistics.log()
-
+    # Map sentences to ids, assign to buckets, compute shard statistics and convert each shard to serialized NDArrays
     data_loader = RawParallelDatasetLoader(buckets=buckets,
                                            eos_id=C.EOS_ID,
                                            pad_id=C.PAD_ID)
 
-    # 3. convert each shard to serialized ndarrays
-    if max_processes == 1:
-        logger.info("Processing shards sequentially.")
-        # Process shards sequentially without using multiprocessing
-        for shard_idx, (shard_sources, shard_targets, shard_stats) in enumerate(shards):
-            save_shard(shard_idx, data_loader, shard_sources, shard_targets,
-                       shard_stats, output_prefix, keep_tmp_shard_files)
-    else:
-        logger.info("Processing shards using %s processes.", max_processes)
-        # Process shards in parallel using max_processes process
-        results = []
-        pool = multiprocessing.pool.Pool(processes=max_processes)
-        for shard_idx, (shard_sources, shard_targets, shard_stats) in enumerate(shards):
-            args = (shard_idx, data_loader, shard_sources, shard_targets,
-                    shard_stats, output_prefix, keep_tmp_shard_files)
-            result = pool.apply_async(save_shard, args=args)
-            results.append(result)
-        pool.close()
-        pool.join()
-
-        for result in results:
-            if not result.successful():
-                logger.error("Process ended in error.")
-                raise RuntimeError("Shard processing failed.")
+
+    # Process shards in parallel using max_processes process
```

Comment:
nit: I'd remove using max_processes process
```suggestion
    # Process shards in parallel
```